contracts/bonding/BondingManager.sol

```diff
@@ -884,21 +835,78 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
+            transcoderPool.updateKey(_transcoder, newStake, address(0), address(0));
+            nextRoundTotalActiveStake = nextRoundTotalActiveStake.add(_amount);
+            lastActiveStakeUpdateRound[_transcoder] = nextRound;
+        } else {
+            // Check if the transcoder is eligible to join the active set in the update round
+            tryToJoinActiveSet(_transcoder, newStake, currentRound);
+        }
+    }
+
+    function decreaseTranscoderTotalStake(address _transcoder, uint256 _amount) internal {
+        if (!transcoderPool.contains(_transcoder)) {
+            return;
+        }
+
+        uint256 newStake = transcoderTotalStake(_transcoder).sub(_amount);
+        uint256 nextRound = roundsManager().currentRound().add(1);
+
+        transcoderPool.updateKey(_transcoder, newStake, address(0), address(0));
+        nextRoundTotalActiveStake = nextRoundTotalActiveStake.sub(_amount);
+        if (transcoderPool.contains(_transcoder)) {
+            lastActiveStakeUpdateRound[_transcoder] = nextRound;
+        } else {
+            lastActiveStakeUpdateRound[_transcoder] = 0;
+        }
+    }
+
+    function tryToJoinActiveSet(address _transcoder, uint256 _totalStake, uint256 _currentRound) internal {
+        uint256 nextRound = _currentRound.add(1);
+        uint256 pendingNextRoundTotalActiveStake = nextRoundTotalActiveStake;
+
+        if (transcoderPool.isFull()) {
+            address lastTranscoder = transcoderPool.getLast();
+            uint256 lastStake = transcoderTotalStake(lastTranscoder);
+
+            // If the pool is full and the transcoder has less stake than the least stake transcoder in the pool
+            // then the transcoder is unable to join the active set for the next round
+            if (_totalStake <= lastStake) {
+                return;
+            }
+
+            // Evict the least stake transcoder from the active set for the next round
+            transcoderPool.remove(lastTranscoder);
+            transcoders[lastTranscoder].earningsPoolPerRound[nextRound].deactivateStake();
+            lastActiveStakeUpdateRound[lastTranscoder] = 0;
+            pendingNextRoundTotalActiveStake = pendingNextRoundTotalActiveStake.sub(lastStake);
+
+            emit TranscoderEvicted(lastTranscoder);
+        }
+
+        transcoderPool.insert(_transcoder, _totalStake, address(0), address(0));
+        transcoders[_transcoder].earningsPoolPerRound[nextRound].activateStake(_totalStake);
+        pendingNextRoundTotalActiveStake = pendingNextRoundTotalActiveStake.add(_totalStake);
+        lastActiveStakeUpdateRound[_transcoder] = nextRound;
+        nextRoundTotalActiveStake = pendingNextRoundTotalActiveStake;
+    }
+
     /**
      * @dev Remove transcoder
      */
     function resignTranscoder(address _transcoder) internal {
+        uint256 transcoderStake = transcoderTotalStake(_transcoder);
+        decreaseTranscoderTotalStake(_transcoder, transcoderStake);
```

Comment:
We don't need this call here since the other statements in this function will remove the address from the active set if needed.